runway/cfngin/exceptions.py

```diff
@@ -250,7 +288,6 @@ def __init__(self, key: str, *args: Any, **kwargs: Any) -> None:
             **kwargs: Arbitrary keyword arguments.
 
         """
-        self.key = key
```

Comment:
this needs to be retained, it contributes to this - https://github.com/rackspace/runway/actions/runs/12190961499/job/34009033373#step:7:196

Reply:
After reverting, add:

```python
    def __reduce__(self) -> tuple[type[Exception], tuple[Any, ...]]:
        """Support for pickling."""
        return self.__class__, (self.key,)
```

This allows the exception to reconstruct itself else the return of this method is essentially `self.__class__, (self.message,)` which is where the doubling in the string is coming from